packages/gatsby/src/schema/infer/inference-metadata.js

```diff
@@ -113,19 +113,47 @@ const getType = (value, key) => {
   }
 }
 
-const updateValueDescriptor = ({
-  nodeId,
-  key,
-  value,
-  operation = `add`,
-  descriptor = {},
-}) => {
+const updateValueDescriptor = (
+  { nodeId, key, value, operation = `add` /*: add | del*/, descriptor = {} },
+  path = []
```

Comment:
other places use `Set` for keeping track if already visited things, was there special reason to use array here?

Reply:
I benchmarked it, this seemed more efficient (although it could be variation). Perhaps because this is an explicit push/pop stack, where the other is arbitrary push.